quickwit/quickwit-serve/src/rest.rs

```diff
@@ -125,10 +131,24 @@ pub(crate) async fn start_rest_server(
         .with(request_counter)
         .recover(recover_fn);
 
+    let warp_service = warp::service(rest_routes);
+    let compression_predicate =
+        DefaultPredicate::new().and(SizeAbove::new(MINIMUM_RESPONSE_COMPRESSION_SIZE));
+
+    let service = ServiceBuilder::new()
+        .layer(
+            CompressionLayer::new()
+                .gzip(true)
+                .compress_when(compression_predicate),
+        )
+        .service(warp_service);
```

Comment:
To clarify why we use tower_http here rather than warp's version.

Warp's compression middleware compresses all responses without any adjustment, making it compress at times we don't need or want, the tower middleware is more adjustable for this purpose.

It also lays the ground work in situations if we want to be able to send compressed bodies to Quickwit via the ingest API in future.